plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/PlayAssetDeliveryUtils.java

```diff
@@ -73,6 +89,16 @@ public static Dictionary convertAssetPackStateToDictionary(AssetPackState assetP
         assetPackState.transferProgressPercentage());
   }
 
+  public static Dictionary convertAssetPackStatesToDictionary(AssetPackStates assetPackStates) {
+    Dictionary returnDict = constructAssetPackStatesDictionary(assetPackStates.totalBytes());
+    Map<String, AssetPackState> packNameStateMap = assetPackStates.packStates();
+    for (Map.Entry<String, AssetPackState> entry : packNameStateMap.entrySet()) {
```

Comment:
Rather than append for every element, could you instead use streams to create a dictionary from `packNameStateMap` and pass that into `constructAssetPackStatesDictionary()`?

Reply:
Modified.